ambry-rest/src/test/java/com.github.ambry.rest/NettyRequestTest.java

```diff
@@ -653,6 +660,11 @@ private void validateRequest(NettyRequest nettyRequest, RestMethod restMethod, S
     assertEquals("Mismatch in uri", uri, nettyRequest.getUri());
     assertNotNull("There should have been a RestRequestMetricsTracker", nettyRequest.getMetricsTracker());
     assertFalse("Should not have been a multipart request", nettyRequest.isMultipart());
+    if (channel.pipeline().get(SslHandler.class) == null) {
+      assertNull("Non-null SSLSession when pipeline does not contain an SslHandler", nettyRequest.getSSLSession());
+    } else {
+      assertNotNull("Null SSLSession when pipeline contains an SslHandler", nettyRequest.getSSLSession());
```

Comment:
any way to check the `SSLSession` is as expected i.e. contains the right certificate, principal etc? Since we generated it, there should be something to compare to?